src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1207,6 +1217,53 @@ private Task WriteStringAsync(string s, bool async)
             return WriteStringAsyncSlow(s, async);
         }
 
+        private Task WriteStringAsync(string s, Encoding? encoding, bool async)
+        {
+            if (encoding is null)
+            {
+                return WriteStringAsync(s, async);
+            }
+
+            // If there's enough space in the buffer to just copy all of the string's bytes, do so.
+            int offset = _writeOffset;
+            int available = _writeBuffer.Length - offset;
+
+            // Fast-path Latin1 and UTF8 as the most common scenarios of non-default encoding
+            const int MaxUtf8BytesPerChar = 3;
```

Comment:
UTF-8 uses a 4-byte max encoding, so this isn't immediately obviously correct. Can you use a comment similar to this one to make it clear?

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.cs#L49-L56

@GrabYourPitchforks do we have some UTF-8 support code in /Common that we could place this constant in?

Reply:
Simplified to just GetMaxByteCount, as it is a very fast method anyway.